python-frontend/src/main/java/org/sonar/plugins/python/api/PythonInputFileContext.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python.api;
+
+import java.io.File;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.plugins.python.api.caching.CacheContext;
+
+public class PythonInputFileContext {
```

Comment:
To make sure that SonarSecurity is able to write the Typeshed UCFGs even when no Python file changed (in which case we shouldn't do any analysis at all, but this won't be the case right now I believe), we'd need to add a `stubFilesSymbols` method here as well. Wdyt @gyula-sallai-sonarsource and @andreaguarino ?